### PR TITLE
NATS Manager Documentation for User

### PR DESCRIPTION
**Pull request by grischperl**
**Description**

Changes proposed in this pull request:

- Remove old Owners and Readme
- Add README with general overview for users
- Add detailed docu of the module purpose, flow and configuration

**Related issue(s)**

#79

### CODE REVIEW

File: docs/user/01-manager.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+### Starting NATS Manager
+
+Upon starting the NATS Manager, the controller (following the [Kubebuilder concept](https://book.kubebuilder.io/architecture.html)) 
+creates, watches and reconciles the following resources:
+   
+   - ConfigMap (cm)
+   - Secret (sc)
+   - Service (sv)
+   - Stateful Set (sfs)
```
Comment (mfaizanse) [inline docs/user/01-manager.md:15]:
```suggestion
   - Stateful Set (sts)
```

File: docs/user/01-manager.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   - DestinationRule (dr)
```
Comment (mfaizanse) [inline docs/user/01-manager.md:16]:
```suggestion
   - DestinationRule (dr, [Istio](https://istio.io))
```

File: docs/user/01-manager.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+   ```mermaid
+    graph LR
+     A(Start NATS manager) -->|Controller| D(Creates, watches & reconciles resources: cm, sc, sv, sfs, dr)
```
Comment (mfaizanse) [inline docs/user/01-manager.md:20]:
```suggestion
     A(Start NATS manager) -->|Controller| D(Creates, watches & reconciles resources: cm, sc, sv, sts, dr)
```

File: docs/user/01-manager.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+   ```mermaid
+   graph LR
+     E(NATS CR changes)-->F(Reconciliation triggered)-->|Controller|G(Resources are adapted to reflect the changes)
+   ```
+
+### Reacting to resource changes
+
+When resources are changed or deleted, the controller reacts by restoring the defaults according to the NATS CR.
+Thus, changes cannot be made directly to the resources, but the NATS CR needs to be edited.
```
Comment (mfaizanse) [inline docs/user/01-manager.md:36]:
```suggestion
Thus, changes cannot be made directly to the resources, but via editing the NATS CR.
```

File: docs/user/02-configuration.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+   kubectl get crd nats.operator.kyma-project.io -o yaml
+   ```
+
+The complete NATS CRD can be found [here](https://github.com/kyma-project/nats-manager/blob/main/config/crd/bases/operator.kyma-project.io_nats.yaml#L1) including
+detailed descriptions for each field.
+
+The NATS CR is used to configure the settings of NATS JetStream. The settings can be edited with the following command:
+
+   ```shell
+   k edit -n kyma-system nats.operator.kyma-project.io
```
Comment (mfaizanse) [inline docs/user/02-configuration.md:16]:
```suggestion
   k edit -n kyma-system nats.operator.kyma-project.io <NATS CR Name>
```

File: docs/user/02-configuration.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+   ```
+
+The CRD is equipped with validation rules and defaulting. The CR is automatically filled with sensible defaults
+that can be overridden and validation rules provide guidance when editing the CR.
```
Comment (mfaizanse) [inline docs/user/02-configuration.md:20]:
```suggestion
that can be overridden. The validation rules provide guidance when editing the CR.
```

File: docs/user/README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# NATS Manager Module
+
+This module ships the NATS Manager, which is responsible for managing the lifecycle of a [NATS JetStream](https://docs.nats.io/nats-concepts/jetstream) deployment.
+It observes the state of the NATS JetStream deployment and reconciles its state according to the desired state.
```
Comment (mfaizanse) [inline docs/user/README.md:4]:
```suggestion
It observes the state of the NATS JetStream resources and reconciles it's state according to the desired state.
```

File: docs/user/01-manager.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
+# NATS Manager
+
+This module ships the NATS Manager.
+
+## Module Lifecycle
+
+### Starting NATS Manager
+
+Upon starting the NATS Manager, the controller (following the [Kubebuilder concept](https://book.kubebuilder.io/architecture.html))
```
Comment (NHingerl) [inline docs/user/01-manager.md:9]:
```suggestion
Upon starting the NATS Manager, the controller (following the [Kubebuilder concept](https://book.kubebuilder.io/architecture.html)) creates, watches and reconciles the following resources:
```

File: docs/user/01-manager.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+creates, watches and reconciles the following resources:
```
Comment (NHingerl) [inline docs/user/01-manager.md:10]:
```suggestion
```

File: docs/user/02-configuration.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
+# Configuration
+
+The CustomResourceDefinition (CRD) `nats.operator.kyma-project.io` is a detailed description to define the NATS custom resource.
```
Comment (NHingerl) [inline docs/user/02-configuration.md:3]:
```suggestion
The CustomResourceDefinition (CRD) `nats.operator.kyma-project.io` describes the the NATS custom resource (CR) in detail.
```

File: docs/user/02-configuration.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+To show the current CRD, run the following command:
+
+   ```shell
+   kubectl get crd nats.operator.kyma-project.io -o yaml
+   ```
+
+The complete NATS CRD can be found [here](https://github.com/kyma-project/nats-manager/blob/main/config/crd/bases/operator.kyma-project.io_nats.yaml#L1) including
```
Comment (NHingerl) [inline docs/user/02-configuration.md:10]:
```suggestion
View the complete [NATS CRD](https://github.com/kyma-project/nats-manager/blob/main/config/crd/bases/operator.kyma-project.io_nats.yaml#L1) including detailed descriptions for each field.
```

File: docs/user/02-configuration.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+detailed descriptions for each field.
```
Comment (NHingerl) [inline docs/user/02-configuration.md:11]:
```suggestion
```

File: docs/user/02-configuration.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+The NATS CR is used to configure the settings of NATS JetStream. The settings can be edited with the following command:
```
Comment (NHingerl) [inline docs/user/02-configuration.md:13]:
```suggestion
The NATS CR configures the settings of NATS JetStream. To edit the settings. run:
```

File: docs/user/02-configuration.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+
+   ```shell
+   kubectl edit -n kyma-system nats.operator.kyma-project.io <NATS CR Name>
+   ```
+
+The CRD is equipped with validation rules and defaulting. The CR is automatically filled with sensible defaults
```
Comment (NHingerl) [inline docs/user/02-configuration.md:19]:
```suggestion
The CRD is equipped with validation rules and defaulting, so the CR is automatically filled with sensible defaults. You can override the defaults. The validation rules provide guidance when you edit the CR.
```
Just guessing

File: docs/user/02-configuration.md [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+that can be overridden. The validation rules provide guidance when editing the CR.
```
Comment (NHingerl) [inline docs/user/02-configuration.md:20]:
```suggestion
```

File: docs/user/02-configuration.md [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+
+## Examples
+
+This project contains several sample CRs to provide some guidance. Each can be applied immediately when [installing](../contributor/installation.md) the NATS Manager.
```
Comment (NHingerl) [inline docs/user/02-configuration.md:24]:
```suggestion
Use the following sample CRs as guidance. Each can be applied immediately when you [install](../contributor/installation.md) the NATS Manager.
```

File: docs/user/README.md [hunk 17]
```diff
@@ -0,0 +1,23 @@
+# NATS Manager Module
+
+This module ships the NATS Manager, which is responsible for managing the lifecycle of a [NATS JetStream](https://docs.nats.io/nats-concepts/jetstream) deployment.
+It observes the state of the NATS JetStream resources and reconciles it's state according to the desired state.
```
Comment (NHingerl) [inline docs/user/README.md:4]:
```suggestion
It observes the state of the NATS JetStream resources and reconciles its state according to the desired state.
```

File: docs/user/README.md [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+NATS is an infrastructure that enables the exchange of data in form of messages. JetStream is a distributed persistence system providing more functionalities
```
Comment (NHingerl) [inline docs/user/README.md:6]:
```suggestion
NATS is an infrastructure that enables the exchange of data in form of messages. JetStream is a distributed persistence system providing more functionalities and higher qualities of service on top of 'Core NATS'.
```

File: docs/user/README.md [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+and higher qualities of service on top of 'Core NATS'.
```
Comment (NHingerl) [inline docs/user/README.md:7]:
```suggestion
```

**Review by mfaizanse — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**